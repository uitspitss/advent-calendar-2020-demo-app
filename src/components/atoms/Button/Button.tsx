import React, { FC, ComponentPropsWithRef } from 'react';
import tw, { styled } from 'twin.macro';

export type ButtonProps = {
  children?: never;
  className?: string;
  /**
   * ラベル
   */
  label: string;
  /**
   * 背景色
   */
  primary?: boolean;
  /**
   * 背景色
   */
  secondary?: boolean;
  /**
   * サイズ
   */
  size?: 'large' | 'small';
} & ComponentPropsWithRef<'button'>;

type Props = ButtonProps;

const Component: FC<Props> = (props) => {
  const { className, label, type } = props;

  return (
    <button
      className={className}
      type={type && type === 'submit' ? 'submit' : 'button'}
    >
      {label}
    </button>
  );
};

const StyledComponent = styled(Component)<
  Pick<Props, 'primary' | 'secondary' | 'size'>
>`
  ${tw`text-base rounded border-none p-2`}

  /* color */
  ${({ primary }) => primary && tw`bg-primary-500 text-gray-200`}
  ${({ secondary }) => secondary && tw`bg-secondary-500 text-gray-200`}

  /* size */
  ${({ size }) => size === 'large' && tw`text-lg`}
  ${({ size }) => size === 'small' && tw`text-sm`}
`;

/**
 * Primary UI component for user interaction
 */
export const Button: FC<ButtonProps> = (props) => (
  <StyledComponent {...props} />
);
