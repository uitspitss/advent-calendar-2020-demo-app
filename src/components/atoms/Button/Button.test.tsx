import React from 'react';
import renderer from 'react-test-renderer';

import { Button } from './Button';

describe('Button', () => {
  test('Snap Shot', () => {
    const tree = renderer.create(<Button label="button" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
