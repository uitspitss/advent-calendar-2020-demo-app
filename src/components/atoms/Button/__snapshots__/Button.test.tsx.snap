// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snap Shot 1`] = `
.emotion-0 {
  font-size: 1rem;
  border-radius: 0.25rem;
  border-style: none;
  padding: 0.5rem;
}

<button
  className="emotion-0 emotion-1"
  type="button"
>
  button
</button>
`;
